### ISSUE TITLE
Checkpoint 2

### ISSUE THREAD

**Issue by PRBergerIII** [PRBergerIII/entJavaIndyProject#5]
@pawaitemadisoncollege I finally have the database set up and some entities happening. I haven't finished mapping all my entities yet, but I'm hoping the worst is behind me in that front. I have the generic DAO working and unit tested. I have some methods commented out for now with notes to delete them if I don't end up using them for anything. Anyway, reflection is [here](https://github.com/PRBergerIII/entJavaIndyProject/blob/main/TimeLog.md#031), I'm hoping to get you another checkpoint tomorrow or friday, and then a marathon through the weekend and early next week. I'll get it done, one way or another.

**Comment by PRBergerIII** [comment]
@pawaitemadisoncollege Added a second tag [here](https://github.com/PRBergerIII/entJavaIndyProject/releases/tag/checkpoint-2.1) which includes a database diagram that I forgot to add earlier, and all the work I did tonight.

**Comment by pawaitemadisoncollege** [comment]
Hey @PRBergerIII I'm finally catching up on Ent Java review work. Sorry for the delay! 

Looks like some great progress here! That's gotta feel really good! DB design and the work so far looks solid overall.

- [ ] There's some redundant code adding records to the database, for example: https://github.com/PRBergerIII/entJavaIndyProject/blob/3043e73744fa9108a0189454ffeb84099e44a5db/src/test/java/com/prberger3/flexregistry/persistence/WishListDaoTest.java#L83-L85 and https://github.com/PRBergerIII/entJavaIndyProject/blob/3043e73744fa9108a0189454ffeb84099e44a5db/src/test/java/com/prberger3/flexregistry/persistence/WishListDaoTest.java#L96-L98.  The unit tests could be simplified by adding these records in the cleandbsql, that way, the unit tests can focus on testing the various conditions and not so much about setting up the db.
- [x] is there a test to confirm that after deleting a user that any associated records are appropriately handled? Essentially, what should happen to the WishLists for that user, and does that happen? Maybe that is the TODO listed in the userdaotest....

**Comment by PRBergerIII** [comment]
@pawaitemadisoncollege 
>is there a test to confirm that after deleting a user that any associated records are appropriately handled? 

There is now! Did one for lists/items too 😄 

Regarding the redundant code, are you saying that it is redundant to have a save or updade and insert check in multiple test classes since they are all working off of the same dao? If so, the reason I did this was partially to make sure that my hibernate mapping in each class was working properly. If you're asking why I repeated that code in particular, it was just that I needed a new object to insert and update in their respective tests, and it was easier to just copy the code rather than make distinct objects. Does that make sense? Or am I misunderstanding the issue?

**Comment by pawaitemadisoncollege** [comment]
Hey Paul,

I was thinking maybe a helper method to create WishListItems if the exact same List were being created. Totally not a big deal!

Good luck on the final push today!

Paula

On May 9, 2022, at 8:11 PM, Paul Berger ***@***.******@***.***>> wrote:



@pawaitemadisoncollege<https://nam10.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fpawaitemadisoncollege&data=05%7C01%7Cpawaite%40madisoncollege.edu%7Cf482c1d8904a438db5fb08da32220ac7%7C33f001466fcc49e9b5687896b3069d44%7C0%7C0%7C637877419033593464%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=GW4tqH2Viyk7uvlF2J4Fn1Efb5NmrPSPc6zGbaLkhDw%3D&reserved=0>

is there a test to confirm that after deleting a user that any associated records are appropriately handled?
There is now!

Regarding the redundant code, are you saying that it is redundant to have a save or updade and insert check in multiple test classes since they are all working off of the same dao? If so, the reason I did this was partially to make sure that my hibernate mapping in each class was working properly. If you're asking why I repeated that code in particular, it was just that I needed a new object to insert and update in their respective tests, and it was easier to just copy the code rather than make distinct objects. Does that make sense? Or am I misunderstanding the issue?
